src/wp-includes/class-wp-block-type.php

```diff
@@ -634,4 +678,112 @@ public function get_uses_context() {
+			 * If this support config has already been stabilized, skip it.
+			 * A stable support key occurring after an experimental key, gets
+			 * stabilized then so that the two configs can be merged effectively.
+			 */
+			if ( isset( $done[ $support ] ) ) {
+				continue;
+			}
+
+			$stable_support_key = $this->experimental_supports_map[ $support ] ?? $support;
+
+			/*
+			 * Use the support's config as is when it's not in need of stabilization.
+			 *
+			 * A support does not need stabilization if:
+			 * - The support key doesn't need stabilization AND
+			 * - Either:
+			 *     - The config isn't an array, so can't have experimental properties OR
+			 *     - The config is an array but has no experimental properties to stabilize.
+			 */
+			if ( $support === $stable_support_key &&
+				( ! is_array( $config ) ||
+					(
+						! isset( $this->experimental_support_properties[ $stable_support_key ] ) &&
+						empty( array_intersect_key( $this->common_experimental_support_properties, $config ) )
+					)
+				)
+			) {
+				$updated_supports[ $support ] = $config;
+				continue;
+			}
+
+			$stabilize_config = function ( $unstable_config, $stable_support_key ) {
+				$stable_config = array();
+				foreach ( $unstable_config as $key => $value ) {
+					// Get stable key from support-specific map, common properties map, or keep original.
+					$stable_key = $this->experimental_support_properties[ $stable_support_key ][ $key ] ??
+							$this->common_experimental_support_properties[ $key ] ??
+							$key;
+
+					$stable_config[ $stable_key ] = $value;
+				}
+				return $stable_config;
+			};
+
+			// Stabilize the config value.
+			$stable_config = is_array( $config ) ? $stabilize_config( $config, $stable_support_key ) : $config;
+
+			/*
+			 * If a plugin overrides the support config with the `register_block_type_args`
+			 * filter, both experimental and stable configs may be present. In that case,
+			 * use the order keys are defined in to determine the final value.
+			 *    - If config is an array, merge the arrays in their order of definition.
+			 *    - If config is not an array, use the value defined last.
+			 *
+			 * The reason for preferring the last defined key is that after filters
+			 * are applied, the last inserted key is likely the most up-to-date value.
+			 * We cannot determine with certainty which value was "last modified" so
+			 * the insertion order is the best guess. The extreme edge case of multiple
+			 * filters tweaking the same support property will become less over time as
+			 * extenders migrate existing blocks and plugins to stable keys.
+			 */
+			if ( $support !== $stable_support_key && isset( $supports[ $stable_support_key ] ) ) {
+				$key_positions      = array_flip( array_keys( $supports ) );
+				$experimental_first =
+				( $key_positions[ $support ] ?? PHP_INT_MAX ) <
+				( $key_positions[ $stable_support_key ] ?? PHP_INT_MAX );
+
+				if ( is_array( $supports[ $stable_support_key ] ) ) {
```

Comment:
I'm not sure but I think the `is_array( $config )` check from [this Gutenberg PR](https://github.com/WordPress/gutenberg/pull/66918/files#diff-76236aad75203b8bb9c5c1bb8c6c6bdd9751d9226f624c4e89327af5c11cb0b4R69) is missing.

So testing using the steps in https://github.com/WordPress/gutenberg/pull/66918, when I set the entire support config to a bool, e.g., `settings.__experimentalBorder = false` using a filter I get the PHP error:

```
Fatal error: Uncaught TypeError: array_merge(): Argument #2 must be of type array, bool given in /var/www/html/wp-includes/class-wp-block-type.php:774
```

Presumably because `$stable_config` is a bool (false).

```suggestion
				if ( is_array( $supports[ $stable_support_key ] ) &&  is_array( $stable_config ) ) {
```

Reply:
Thanks for flagging this, I'll investigate it.

The border support stabilization's approach and code was changed by the final PR stabilizing the common experimental flags. There's a good chance that last PR introduced this bug to Gutenberg as well. I'll need to looking into both.

Reply:
At a quick glance, this backport does reflect the Gutenberg code in its current state. Both will need to be updated to handle a scenario where one of the top-level keys (experimental or stable) is a boolean.

I'll update this shortly and spin up a quick PR to fix Gutenberg.

Reply:
A quick PR for the Gutenberg side of this is up in https://github.com/WordPress/gutenberg/pull/67552.

Reply:
I've brought across the fix from https://github.com/WordPress/gutenberg/pull/67552 in https://github.com/WordPress/wordpress-develop/pull/7069/commits/ef6959b2c7b2a56ceca77340174b7b5317fe5639. I'm out of time today but will give it more of a test tomorrow.